### DISCUSSION TITLE
Egui for Julia using JuliaWGPU

### DISCUSSION THREAD

**Discussion by ANIL80531** [JuliaWGPU/WGPUgfx.jl#20] [Q&A]
Hey I'm thinking of making an egui.rs like GUI library for julia which can support WGPU and run on browsers. I am not able to find a good resource for you JuliaWGPU library. Can you provide a good resource which helped you build this pkg

**Comment by arhik** [comment]
There are many things that needs to be done to get there. WASM compilation support is still a heavy lifting for browser support. If you can afford you can start working on desktop versions and then later adapt for browser related updates. We started working on GUI library and would like something like https://lib.rs/crates/iced_wgpu in our approach.

**Comment by arhik** [comment]
> Can you provide a good resource which helped you build this pkg. 

Browse to gen folder in WGPUNative.jl repo and you will find code that's self-explanatory. I will document the process in due time which you can follow.
These two might help you:
1. https://pkgdocs.julialang.org/v1/artifacts/
2. https://juliainterop.github.io/Clang.jl/stable/generator/

**Comment by VarLad** [comment]
@ANIL80531 Recent examples in WGPUgfx.jl and the package SimpleImGui.jl might be a very good start to recreate egui in Julia. :)
